modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/related/models/test/ObjectRelatedModelsProviderTest.java

```diff
@@ -1125,11 +1069,271 @@ private void _testSystemObjectEntry1toMObjectRelatedModelsProviderImpl()
+		_assertSearchRelatedModels(primaryKey, "Entry", 2);
+	}
+
+	private void _testUserSystemObjectEntry1toMObjectRelatedModelsProviderImpl()
+		throws Exception {
+
+		long[] userIds = _addUsers(3);
+
+		ObjectDefinition userSystemObjectDefinition =
+			_objectDefinitionLocalService.fetchObjectDefinitionByClassName(
+				TestPropsValues.getCompanyId(), User.class.getName());
+
+		_addObjectRelationship(
+			_objectDefinition2, userSystemObjectDefinition,
+			ObjectRelationshipConstants.DELETION_TYPE_PREVENT,
+			ObjectRelationshipConstants.TYPE_ONE_TO_MANY);
+
+		ObjectEntry objectEntry1 = _addObjectEntry(
+			_objectDefinition2.getObjectDefinitionId(), Collections.emptyMap());
+
+		User user = _userLocalService.getUser(userIds[2]);
+
+		_testSystemObjectEntry1toMObjectRelatedModels(
+			user.getFirstName(), objectEntry1.getObjectEntryId(), userIds,
+			userSystemObjectDefinition.getObjectDefinitionId());
+
+		// Object relationship deletion type cascade
+
+		_updateObjectRelationship(
+			ObjectRelationshipConstants.DELETION_TYPE_CASCADE);
+
+		_objectEntryLocalService.deleteObjectEntry(objectEntry1);
+
+		AssertUtils.assertFailure(
+			NoSuchUserException.class,
+			"No User exists with the primary key " + userIds[1],
+			() -> _userLocalService.getUser(userIds[1]));
+		AssertUtils.assertFailure(
+			NoSuchUserException.class,
+			"No User exists with the primary key " + userIds[2],
+			() -> _userLocalService.getUser(userIds[2]));
+
+		// Object relationship deletion type disassociate
+
+		_testSystemObjectEntry1toMDeletionTypeDisassociate(
+			userIds[0], userSystemObjectDefinition.getObjectDefinitionId());
+
+		Assert.assertNotNull(_userLocalService.fetchUser(userIds[0]));
+
+		// Object relationship deletion type prevent
+
+		_testSystemObjectEntry1toMDeletionTypePrevent(
+			userIds[0], userSystemObjectDefinition.getObjectDefinitionId());
 
 		_objectRelationshipLocalService.deleteObjectRelationship(
 			_objectRelationship);
 	}
 
+	private void _testUserSystemObjectEntryMtoMObjectRelatedModelsProviderImpl()
+		throws Exception {
+
+		long[] userIds = _addUsers(3);
+
+		ObjectDefinition userSystemObjectDefinition =
+			_objectDefinitionLocalService.fetchObjectDefinitionByClassName(
+				TestPropsValues.getCompanyId(), User.class.getName());
+
+		_addObjectRelationship(
+			_objectDefinition1, userSystemObjectDefinition,
+			ObjectRelationshipConstants.DELETION_TYPE_PREVENT,
+			ObjectRelationshipConstants.TYPE_MANY_TO_MANY);
+
+		ObjectEntry objectEntry1 = _addObjectEntry(
+			_objectDefinition1.getObjectDefinitionId(), Collections.emptyMap());
+
+		User user = _userLocalService.getUser(userIds[1]);
+
+		_testSystemObjectEntryMtoMObjectRelatedModels(
+			user.getFirstName(), objectEntry1.getObjectEntryId(), userIds);
+
+		// Object relationship deletion type cascade
+
+		_updateObjectRelationship(
+			ObjectRelationshipConstants.DELETION_TYPE_CASCADE);
+
+		_userLocalService.deleteUser(userIds[1]);
```

Comment:
I think we can encapsulate this logic in private method and then reuse twice here:

```
_userLocalService.deleteUser(userIds[1]);

Assert.assertNotNull(
	_objectEntryLocalService.fetchObjectEntry(
		objectEntry1.getObjectEntryId()));

AssertUtils.assertFailure(
	NoSuchUserException.class,
	"No User exists with the primary key " + userIds[1],
	() -> _userLocalService.getUser(userIds[1]));
```
			
I understood that it basically delete the entry, then `assertNotNull` and `assertFailure`. What do you think?
You would need to pass as an argument something like: `(Class<?> clazz, UnsafeFunction<T, R, E extends Throwable> deleteUnsafeFunction, String message, UnsafeRunnable<Exception> unsafeRunnable)`
Please see this example: [AccountRoleLocalServiceTest#_testDeleteAccountRole](https://github.com/liferay/liferay-portal/blob/53a8c5dd134b7cf6e6ff9efd9a97db1c629c901c/modules/apps/account/account-test/src/testIntegration/java/com/liferay/account/service/test/AccountRoleLocalServiceTest.java#L552C1-L555)